BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/OperationHolder.cs

```diff
@@ -105,13 +105,18 @@ protected virtual void Dispose(bool disposing)
 
                             currentActivity?.Stop();
 
-                            if (this.originalActivity != null && 
-                                Activity.Current != this.originalActivity && 
+                            if (this.originalActivity != null &&
+                                Activity.Current != this.originalActivity &&
                                 this.originalActivity is Activity activity)
                             {
                                 Activity.Current = activity;
                             }
-                        });
+                        }
+                        catch (Exception)
+                        {
+                            CoreEventSource.Log.ActivityNotAvailable();
```

Comment:
I expect that we should log the exception here as well.

Reply:
thanks for reviewing. just added the parameter to save it!